compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1569,7 +1570,7 @@ class Namer { typer: Typer =>
             if pclazz.is(Final) then
               report.error(ExtendFinalClass(cls, pclazz), cls.srcPos)
             else if pclazz.isEffectivelySealed && pclazz.associatedFile != cls.associatedFile then
-              if pclazz.is(Sealed) then
+              if pclazz.is(Sealed) && !pclazz.is(JavaDefined) then
```

Comment:
this would seem to allow extending a sealed Java class in a Scala file, perhaps use `!isJava` instead which tests the compilation unit. Or could it be legitimate to define a permitted sub class in Scala?
```suggestion
              if pclazz.is(Sealed) && !isJava then
```

Reply:
maybe it could be interesting to add this as a test case